src/enrichers/idcookie.ts

```diff
@@ -0,0 +1,23 @@
+import { WrappedReadOnlyStorageHandler } from '../handlers/storage-handler'
+import { Enricher, IdCookieConfig } from '../types'
+
+type Input = { idCookie?: IdCookieConfig, peopleVerifiedId?: string }
+type Output = { resolvedIdCookie: string | null }
+
+export function enrichIdCookie(
+  storageHandler: WrappedReadOnlyStorageHandler
+): Enricher<Input, Output> {
+  return state => {
+    let resolvedIdCookie: string | null
+
+    if (state.idCookie?.mode === 'provided' && state.idCookie?.strategy === 'cookie' && typeof state.idCookie?.name === 'string') {
```

Comment:
We can make `'generated', 'provided', 'cookie', 'localStorage' ` constants and use them, WDYT

Reply:
Not really any benefit when using typescript. It is already typed as `"provided" | "generated"` and will complain if you try to compare it to any other string